### PR TITLE
Added functionality to allow custom S3 endpoints

### PR DESCRIPTION
**Pull request by jinhong-**
**Proposed changes**:
- #444   

**Status**:
- [x] ready for code review
- [ ] there are tests for the functionality
- [x] documentation updated
- [ ] changelog updated

### CODE REVIEW

File: rasa_nlu/persistor.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def get_persistor(config):
         raise KeyError("No persistent storage specified. Supported values are {}".format(", ".join(['aws', 'gcs'])))
 
     if config['storage'] == 'aws':
-        return AWSPersistor(config['path'], config['aws_region'], config['bucket_name'])
+        return AWSPersistor(config['path'], config['aws_region'], config['bucket_name'], config.as_dict().get('aws_endpoint_url'))
```
Comment (tmbo) [inline rasa_nlu/persistor.py]: The configuration value needs to be added to `config_defaults.json` as well as into the dict `DEFAULT_CONFIG` in `config.py`. (probably with the default value `None`). 

After that is done, the access here should be `config['aws_endpoint_url']`

File: rasa_nlu/persistor.py [hunk 2]
```diff
@@ -47,11 +47,11 @@ def fetch_and_extract(self, filename):
 class AWSPersistor(Persistor):
     """Store models on S3 and fetch them when needed instead of storing them on the local disk."""
 
-    def __init__(self, data_dir, aws_region, bucket_name):
+    def __init__(self, data_dir, aws_region, bucket_name, endpoint_url):
         # type: (Text, Text, Text) -> None
         Persistor.__init__(self)
         self.data_dir = data_dir
-        self.s3 = boto3.resource('s3', region_name=aws_region)
+        self.s3 = boto3.resource('s3', region_name=aws_region, endpoint_url=endpoint_url)
```
Comment (tmbo) [inline rasa_nlu/persistor.py]: Does this work if `edpoint_url` is `None`?

File: rasa_nlu/persistor.py (hunk 1 above)
Comment (jinhong-) [inline rasa_nlu/persistor.py]: Alright will change that

File: rasa_nlu/persistor.py (hunk 2 above)
Comment (jinhong-) [inline rasa_nlu/persistor.py]: image

Any AWS endpoint for me to test on? I think it should work based on the documents here

Comment (jinhong-) [inline rasa_nlu/persistor.py]: http://boto3.readthedocs.io/en/latest/_modules/boto3/session.html

Comment (tmbo) [inline rasa_nlu/persistor.py]: yes looks good :+1:

**Comment by jinhong-** [comment]
All good now

**Comment by wrathagom** [comment]
@tmbo is this one ready to go?

**Comment by tmbo** [comment]
Yes. @jinhong- Thanks a lot for your contribution 🎉

**Pull request merged**